src/state/codeReceiver.js

```diff
@@ -0,0 +1,48 @@
+import { Observable } from 'rxjs/Rx';
+
+import { dispatchAction, getState } from '../reducers/public';
+import { addNode } from '../reducers/picklist';
+import { getPath } from '../reducers/helper';
+import { getMuseumId } from '../reducers/auth';
+import { hashHistory } from 'react-router';
+
+import Config from '../config';
+
+const UUID_REGEX = /^[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$/i;
```

Comment:
FYI: I believe there are some rules to how this numeric representation of old barcodes is built up. So it might be possible (in the scanner code below) to represent that as a regex as well.